js_modules/dagster-ui/packages/ui-core/src/asset-data/AssetLiveDataProvider.tsx

```diff
@@ -1,92 +1,129 @@
-import {ApolloClient, gql, useApolloClient} from '@apollo/client';
 import uniq from 'lodash/uniq';
-import React from 'react';
+import React, {useCallback, useMemo, useRef} from 'react';
 
+import {AssetBaseData, __resetForJest as __resetBaseData} from './AssetBaseDataProvider';
 import {
-  AssetGraphLiveQuery,
-  AssetGraphLiveQueryVariables,
-} from './types/AssetLiveDataProvider.types';
+  AssetStaleStatusData,
+  __resetForJest as __resetStaleData,
+} from './AssetStaleStatusDataProvider';
 import {SUBSCRIPTION_MAX_POLL_RATE} from './util';
 import {observeAssetEventsInRuns} from '../asset-graph/AssetRunLogObserver';
-import {
-  LiveDataForNode,
-  buildLiveDataForNode,
-  tokenForAssetKey,
-  tokenToAssetKey,
-} from '../asset-graph/Utils';
+import {LiveDataForNodeWithStaleData, tokenForAssetKey} from '../asset-graph/Utils';
 import {AssetKeyInput} from '../graphql/types';
-import {liveDataFactory} from '../live-data-provider/Factory';
 import {LiveDataPollRateContext} from '../live-data-provider/LiveDataProvider';
 import {LiveDataThreadID} from '../live-data-provider/LiveDataThread';
 import {useDidLaunchEvent} from '../runs/RunUtils';
 
-function makeFactory() {
-  return liveDataFactory(
-    () => {
-      return useApolloClient();
-    },
-    async (keys, client: ApolloClient<any>) => {
-      const {data} = await client.query<AssetGraphLiveQuery, AssetGraphLiveQueryVariables>({
-        query: ASSETS_GRAPH_LIVE_QUERY,
-        fetchPolicy: 'network-only',
-        variables: {
-          assetKeys: keys.map(tokenToAssetKey),
-        },
-      });
-      const nodesByKey = Object.fromEntries(
-        data.assetNodes.map((node) => [tokenForAssetKey(node.assetKey), node]),
-      );
-
-      const liveDataByKey = Object.fromEntries(
-        data.assetsLatestInfo.map((assetLatestInfo) => {
-          const id = tokenForAssetKey(assetLatestInfo.assetKey);
-          return [id, buildLiveDataForNode(nodesByKey[id]!, assetLatestInfo)];
-        }),
-      );
-      return liveDataByKey;
-    },
-  );
-}
-export const factory = makeFactory();
-
 export function useAssetLiveData(assetKey: AssetKeyInput, thread: LiveDataThreadID = 'default') {
-  return factory.useLiveDataSingle(tokenForAssetKey(assetKey), thread);
+  const key = tokenForAssetKey(assetKey);
+  const {
+    liveData: staleData,
+    refresh: refreshStaleData,
+    refreshing: staleDataRefreshing,
+  } = AssetStaleStatusData.useLiveDataSingle(key, thread);
+  const {
+    liveData: baseData,
+    refresh: refreshBaseData,
+    refreshing: baseDataRefreshing,
+  } = AssetBaseData.useLiveDataSingle(key, thread);
+
+  const refresh = useCallback(() => {
+    refreshBaseData();
+    refreshStaleData();
+  }, [refreshBaseData, refreshStaleData]);
+  const refreshing = baseDataRefreshing || staleDataRefreshing;
+
+  if (baseData && staleData) {
```

Comment:
Hmm, the fact that we don't show `base` until we have `stale` means that the primary value of caching them separately and composing client-side is to avoid re-fetching `base` data unnecessarily.

Eg: we're choosing to fetch:

A: base 
+ 
B: stale

Instead of:

A: base 
OR 
B: base + stale

The second one is obviously more wasteful but it'd be simpler because we could use a templating pattern instead of a compositional pattern on the client-side.  Is the base data still costly with the new backend improvements? :-o

Reply:
>Hmm, the fact that we don't show base until we have stale
Yeah we could allow showing base before the stale data arrives (though in practice I'm seeing the stale data come before the base) but that would need more granular loading states than what we have now so I omitted that (for now).